hrm-domain/hrm-service/migrations/20230926214506-create-client-profiles-tables.js

```diff
@@ -0,0 +1,305 @@
+      )
+    `);
+    console.log("Table 'Identifiers' created.");
+    await queryInterface.sequelize.query(`
+      ALTER TABLE IF EXISTS public."Identifiers" OWNER to hrm;
+    `);
+    console.log("Table 'Identifiers' ownership altered.");
+    await queryInterface.sequelize.query(`
+    CREATE INDEX IF NOT EXISTS "Identifiers_identifier_accountSid" ON public."Identifiers" USING btree
+      ("identifier" ASC NULLS LAST, "accountSid" COLLATE pg_catalog."default" ASC NULLS LAST)
+    `);
+    console.log(
+      "Index 'fki_ProfilesToProfileFlags_profileId_accountSid_Profiles_id_accountSid_fk' created.",
+    );
+
+    // ProfileFlags
+    await queryInterface.sequelize.query(`
+      CREATE SEQUENCE IF NOT EXISTS public."ProfileFlags_id_seq"
+        INCREMENT 1
+        START 1
+        MINVALUE 1
+        MAXVALUE 9223372036854775807
+        CACHE 1;
+    `);
+    console.log('Sequence "ProfileFlags_id_seq" created');
+    await queryInterface.sequelize.query(`
+    ALTER SEQUENCE public."ProfileFlags_id_seq" OWNER TO hrm
+    `);
+    console.log("Sequence 'ProfileFlags_id_seq' ownership altered.");
+    await queryInterface.sequelize.query(`
+      CREATE TABLE IF NOT EXISTS public."ProfileFlags"
+      (
+        id integer NOT NULL DEFAULT nextval('"ProfileFlags_id_seq"'::regclass),
+        "name" text COLLATE pg_catalog."default" NOT NULL,
+        "accountSid" text COLLATE pg_catalog."default" NOT NULL,
+        "createdAt" timestamp with time zone NOT NULL,
+        "updatedAt" timestamp with time zone NOT NULL,
+        CONSTRAINT "ProfileFlags_pkey" PRIMARY KEY ("id", "accountSid")
+      )
+    `);
+    console.log("Table 'ProfileFlags' created.");
+    await queryInterface.sequelize.query(`
+      ALTER TABLE IF EXISTS public."ProfileFlags" OWNER to hrm;
+    `);
+    console.log("Table 'ProfileFlags' ownership altered.");
+
+    //
+    // TODO: if we populate the ProfileFlags table, we'll need accountSid, which we have now,
+    //       but we wont when a new helpline is added.
+    //       Do we have a plan to "auto-populate" this table on new accounts?
+    //      Wouldn't be better to let each helpline populate this when they need it?
+    //
+
+    // ProfilesToProfileFlags
+    await queryInterface.sequelize.query(`
+      CREATE TABLE IF NOT EXISTS public."ProfilesToProfileFlags"
+      (
+        "profileId" integer NOT NULL,
+        "profileFlagId" integer NOT NULL,
+        "accountSid" text COLLATE pg_catalog."default" NOT NULL,
+        "createdAt" timestamp with time zone NOT NULL,
+        "updatedAt" timestamp with time zone NOT NULL,
+        CONSTRAINT "ProfilesToProfileFlags_pkey" PRIMARY KEY ("profileId", "profileFlagId", "accountSid"),
+        CONSTRAINT "ProfilesToProfileFlags_profileId_Profiles_id_fk" FOREIGN KEY ("profileId", "accountSid")
+          REFERENCES public."Profiles" (id, "accountSid") MATCH SIMPLE
+          ON UPDATE CASCADE
+          ON DELETE CASCADE,
```

Comment:
Change to "nullify".

Reply:
@robert-bo-davis I confirmed that this works as intended:
If Profile A and a Flag 1, and a relationship exists between A to 1 via the relations table, deleting either A or 1 deletes the relation record. The other way around is not true: deleted the relation A-1 does NOT deletes A from Profiles nor 1 from Flags.